frontends/web/src/routes/settings/components/device-settings/firmware-setting.tsx

```diff
@@ -71,11 +72,11 @@ const FirmwareSetting = ({ deviceID, versionInfo, asButton = false }: TProps) =>
         />
       )}
       <UpgradeDialog
-        open={dialogOpen && canUpgrade}
+        open={firmwareUpdateDialogOpen && canUpgrade}
         versionInfo={versionInfo}
         confirming={confirming}
         onUpgradeFirmware={handleUpgradeFirmware}
-        onClose={() => setDialogOpen(false)}
+        onClose={() => setFirmwareUpdateDialogOpen(false)}
```

Comment:
> Also control firmware update dialog from AppContext through
`firmwareUpdateDialogOpen` - so it's controllable from other
pages such as the new sell page.

would be very nice if this was a separate commit.

Reply:
Hehe, okay done :)